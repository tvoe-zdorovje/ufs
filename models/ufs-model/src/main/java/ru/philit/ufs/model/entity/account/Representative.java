package ru.philit.ufs.model.entity.account;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import ru.philit.ufs.model.entity.common.ExternalEntity;

/**
 * Представитель корпоративного бизнеса.
 */
@EqualsAndHashCode(of = {"id"}, callSuper = false)
@ToString
@Getter
@Setter
public class Representative extends ExternalEntity {

  private String id;
  private String inn;
  private String firstName;
  private String lastName;
  private String patronymic;
  private String phoneNumWork;
  private String phoneNumMobile;
  private String email;
  private String address;
  private String postindex;
  private Date birthDate;
  private Sex sex;
  private String placeOfBirth;
  private boolean resident;
  private List<IdentityDocument> identityDocuments;
  private List<LegalEntity> legalEntities;

  public Representative() {
    this.identityDocuments = new ArrayList<>();
    this.legalEntities = new ArrayList<>();
  }
}
