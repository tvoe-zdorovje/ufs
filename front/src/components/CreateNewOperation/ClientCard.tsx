import React from 'react'
import {
  Col,
  Input,
  Row,
  Checkbox
} from 'antd'

import Section from 'components/Section/Section'
import { Representative } from 'interfaces/dto/Representative.interface'

const styles = require('./CreateNewOperation.less')

export default (props: Representative) => {
  const disabledInput = value => (
    <Input disabled value={value} />
  )

  const label = (value) => (<span className={styles.label}>{value}</span>)

  return (
    <div style={{ minWidth: 300 }}>
      <Section header="">
        <Row>
          <Col span={7}>
            {label('Фамилия')}
          </Col>
          <Col span={16}>
            {disabledInput(props.lastName)}
          </Col>
        </Row>
        <Row>
          <Col span={7}>
            {label('Имя')}
          </Col>
          <Col span={16}>
            {disabledInput(props.firstName)}
          </Col>
        </Row>
        <Row>
          <Col span={7}>
            {label('Отчество')}
          </Col>
          <Col span={16}>
            {disabledInput(props.patronymic)}
          </Col>
        </Row>
        <Row>
          <Col span={7}>
            {label('Дата рождения')}
          </Col>
          <Col span={16}>
            {disabledInput(props.birthDate)}
          </Col>
        </Row>
        <Row>
          <Col span={7}>
            {label('Тип ДУЛ')}
          </Col>
          <Col span={16}>
            {disabledInput(props.document.type)}
          </Col>
        </Row>
        <Row>
          <Col span={7}>
            {label('Серия')}
          </Col>
          <Col span={7}>
            {disabledInput(props.document.series)}
          </Col>
          <Col span={4}>
            {label('Номер')}
          </Col>
          <Col span={5}>
            {disabledInput(props.document.number)}
          </Col>
        </Row>
        <Row>
          <Col span={7}>
            {label('Место рождения')}
          </Col>
          <Col span={16}>
            {disabledInput(props.birthPlace)}
          </Col>
        </Row>
        <Row>
          <Col span={7}>
            {label('Дата выдачи')}
          </Col>
          <Col span={16}>
            {disabledInput(props.document.issuedDate)}
          </Col>
        </Row>
        <Row>
          <Col span={7}>
            {label('Выдан (орган)')}
          </Col>
          <Col span={16}>
            {disabledInput(props.document.issuedBy)}
          </Col>
        </Row>
        <Row>
          <Col span={7}>
            {label('ИНН/КИО')}
          </Col>
          <Col span={16}>
            {disabledInput(props.inn)}
          </Col>
        </Row>
        <Row>
          <Col span={7}>
            {label('Индекс')}
          </Col>
          <Col span={16}>
            {disabledInput(props.postcode)}
          </Col>
        </Row>
        <Row>
          <Col span={7}>
            {label('Адрес')}
          </Col>
          <Col span={16}>
            {disabledInput(props.address)}
          </Col>
        </Row>
        <Row>
          <Col span={7}>
            {label('Резидент')}
          </Col>
          <Col span={16}>
            <Checkbox checked={props.resident} />
          </Col>
        </Row>
      </Section>
    </div>
  )
}
