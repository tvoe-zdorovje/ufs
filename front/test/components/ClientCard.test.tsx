import React from 'react'
import {
  render,
  shallow
} from 'enzyme'
import toJson from 'enzyme-to-json'
import { Input } from 'antd'

import { Representative } from 'interfaces/dto/Representative.interface'
import ClientCard from 'components/CreateNewOperation/ClientCard'

describe('Компонент ClientCard', () => {
  const position: string = 'position'
  const client: Representative = {
    inn: '',
    lastName: 'ln',
    firstName: '',
    birthDate: '',
    birthPlace: '',
    resident: true,
    document: {
      type: '',
      series: '',
      number: '',
      issuedBy: '',
      issuedDate: '',
    },
  }

  const wrapper = shallow(<ClientCard {...client} />)
  const renderWrapper = render(<ClientCard {...client} />)

  it('использует свойства <ClientCard />', () => {
    expect(wrapper.find(Input).first().prop('value')).toBe(client.lastName)
  })

  it('отображает поля карточки: <Input />', () => {
    expect(wrapper.find(Input)).toHaveLength(13)
  })

  it('совпадает со своим снимком', () => {
    expect(toJson(renderWrapper)).toMatchSnapshot()
  })
})



