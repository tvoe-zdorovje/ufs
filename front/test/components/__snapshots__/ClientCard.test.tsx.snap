// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент ClientCard совпадает со своим снимком 1`] = `
<div
  style="min-width:300px;"
>
  <section
    class="section"
  >
    <div
      class="header"
    />
    <div
      class="container"
    >
      <div
        class="ant-row"
      >
        <div
          class="ant-col-7"
        >
          <span
            class="label"
          >
            Фамилия
          </span>
        </div>
        <div
          class="ant-col-16"
        >
          <input
            class="ant-input"
            disabled=""
            type="text"
            value="ln"
          />
        </div>
      </div>
      <div
        class="ant-row"
      >
        <div
          class="ant-col-7"
        >
          <span
            class="label"
          >
            Имя
          </span>
        </div>
        <div
          class="ant-col-16"
        >
          <input
            class="ant-input"
            disabled=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="ant-row"
      >
        <div
          class="ant-col-7"
        >
          <span
            class="label"
          >
            Отчество
          </span>
        </div>
        <div
          class="ant-col-16"
        >
          <input
            class="ant-input"
            disabled=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="ant-row"
      >
        <div
          class="ant-col-7"
        >
          <span
            class="label"
          >
            Дата рождения
          </span>
        </div>
        <div
          class="ant-col-16"
        >
          <input
            class="ant-input"
            disabled=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="ant-row"
      >
        <div
          class="ant-col-7"
        >
          <span
            class="label"
          >
            Тип ДУЛ
          </span>
        </div>
        <div
          class="ant-col-16"
        >
          <input
            class="ant-input"
            disabled=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="ant-row"
      >
        <div
          class="ant-col-7"
        >
          <span
            class="label"
          >
            Серия
          </span>
        </div>
        <div
          class="ant-col-7"
        >
          <input
            class="ant-input"
            disabled=""
            type="text"
            value=""
          />
        </div>
        <div
          class="ant-col-4"
        >
          <span
            class="label"
          >
            Номер
          </span>
        </div>
        <div
          class="ant-col-5"
        >
          <input
            class="ant-input"
            disabled=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="ant-row"
      >
        <div
          class="ant-col-7"
        >
          <span
            class="label"
          >
            Место рождения
          </span>
        </div>
        <div
          class="ant-col-16"
        >
          <input
            class="ant-input"
            disabled=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="ant-row"
      >
        <div
          class="ant-col-7"
        >
          <span
            class="label"
          >
            Дата выдачи
          </span>
        </div>
        <div
          class="ant-col-16"
        >
          <input
            class="ant-input"
            disabled=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="ant-row"
      >
        <div
          class="ant-col-7"
        >
          <span
            class="label"
          >
            Выдан (орган)
          </span>
        </div>
        <div
          class="ant-col-16"
        >
          <input
            class="ant-input"
            disabled=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="ant-row"
      >
        <div
          class="ant-col-7"
        >
          <span
            class="label"
          >
            ИНН/КИО
          </span>
        </div>
        <div
          class="ant-col-16"
        >
          <input
            class="ant-input"
            disabled=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="ant-row"
      >
        <div
          class="ant-col-7"
        >
          <span
            class="label"
          >
            Индекс
          </span>
        </div>
        <div
          class="ant-col-16"
        >
          <input
            class="ant-input"
            disabled=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="ant-row"
      >
        <div
          class="ant-col-7"
        >
          <span
            class="label"
          >
            Адрес
          </span>
        </div>
        <div
          class="ant-col-16"
        >
          <input
            class="ant-input"
            disabled=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="ant-row"
      >
        <div
          class="ant-col-7"
        >
          <span
            class="label"
          >
            Резидент
          </span>
        </div>
        <div
          class="ant-col-16"
        >
          <label
            class="ant-checkbox-wrapper"
          >
            <span
              class="ant-checkbox ant-checkbox-checked"
            >
              <input
                checked=""
                class="ant-checkbox-input"
                type="checkbox"
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
          </label>
        </div>
      </div>
    </div>
  </section>
</div>
`;
